site/_guides/gateway-api.md

```diff
@@ -0,0 +1,118 @@
+expose an HTTP application using an HTTPRoute resource.
+
+### Prerequisites
+The following prerequisites must be met before using Gateway API with Contour:
+
+- A working [Kubernetes][2] cluster. Refer to the [compatibility matrix][3] for cluster version requirements.
+- The [kubectl][4] command-line tool, installed and configured to access your cluster.
+
+### Using Gateway API with Contour Operator
+
+The preferred method for using Gateway API is with [Contour Operator][5]. This is because the GatewayClass and Gateway
+resources should be managed by active controllers that are implemented by the operator. Refer to the [contour][6] and
+[operator][7] designs for additional background on the gateway API implementation.
+
+Run the operator:
+```shell
+$ kubectl apply -f https://projectcontour.io/quickstart/operator.yaml
+```
+This command creates:
+
+- Namespace `contour-operator` to run the operator.
+- Operator and Contour CRDs.
+- Operator RBAC resources for the operator.
+- A Deployment to manage the operator.
+- A Service to front-end the operator’s metrics endpoint.
+
+Create the Gateway API resources:
+```shell
+$ kubectl apply -f https://projectcontour.io/quickstart/gateway.yaml
+```
+This command creates:
+
+- Namespace `projectcontour` to run the Gateway and child resources, i.e. Envoy DaemonSet.
+- A Contour custom resource named `contour-gateway-sample` in the operator's namespace. This resource exposes
+  infrastructure-specific configuration and is referenced by the GatewayClass.
+- A GatewayClass named `sample-gatewayclass` that abstracts the infrastructure-specific configuration from Gateways.
+- A Gateway named `contour` in namespace `projectcontour`. This gateway will serve the test application through routing
+  rules deployed in the next step.
+
+### Testing the Gateway API
+
+Run the test application:
+```shell
+$ kubectl apply -f https://projectcontour.io/quickstart/kuard.yaml
+```
+This command creates:
+
+- A Deployment named `kuard` in namespace `projectcontour` to run kuard as the test application.
+- A Service named `kuard` in namespace `projectcontour` to expose the kuard application on TCP port 80.
+- An HTTPRoute named `kuard` in namespace `projectcontour` to route requests for "*.projectcontour.io" to the kuard
+  service.
+
+Verify the kuard resources are available:
+```shell
+$ kubectl get po,svc,httproute -n projectcontour -l app=kuard
+NAME                         READY   STATUS    RESTARTS   AGE
+pod/kuard-798585497b-78x6x   1/1     Running   0          21s
+pod/kuard-798585497b-7gktg   1/1     Running   0          21s
+pod/kuard-798585497b-zw42m   1/1     Running   0          21s
+
+NAME            TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)   AGE
+service/kuard   ClusterIP   172.30.168.168   <none>        80/TCP    21s
+
+NAME                                  HOSTNAMES
+httproute.networking.x-k8s.io/kuard   ["*.projectcontour.io"]
```

Comment:
Can we actually maybe modify the example to not use a wildcard and rather a specific hostname? Since the actual implications of a wildcard hostname are somewhat nuanced and advertising that we support this already is a bit misleading

Reply:
That is a good catch, I agree.

Reply:
@sunjayBhatia https://github.com/projectcontour/contour-operator/pull/220 is required for me to update the hostname.

Reply:
@danehans merged!